pkg/pillar/cmd/downloader/syncop.go

```diff
@@ -323,12 +323,17 @@ func sourceFailureError(ip, ifname, url string, err error) {
 
 func getDatastoreCredential(ctx *downloaderContext,
 	dst types.DatastoreConfig) (zconfig.CredentialBlock, error) {
-	cred := utils.PrepareCipherCred(dst.ApiKey, dst.Password)
-	status, cred, err := utils.GetCipherCredentials(agentName,
-		dst.CipherBlockStatus, cred)
-	if status.IsCipher && len(status.Error) == 0 {
+	var cred zconfig.CredentialBlock
+	var status types.CipherBlockStatus
+	var err error
+	status, cred, err = utils.GetCipherCredentials(agentName,
+		dst.CipherBlockStatus)
+	if status.IsCipher && err == nil {
 		log.Infof("%s, cipherblock decryption successful\n", dst.Key())
+	} else {
+		log.Infof("%s, datastore config cipherblock decryption unsuccessful\n", dst.Key())
+		cred = utils.PrepareCipherCred(dst.ApiKey, dst.Password)
```

Comment:
Should we add a notion of this failing (for the case when we have no password) to handle the future case when the cleartext is no longer included and only the ciphertext?
Note that we can't check for having an empty apikey/password since that can be normal.
But we can check for 1) non-empty cipher 2) cipher failed and 3) empty cleartext. In that case it doesn't make sense to proceed with the empty key/password.

Reply:
FWIW We can do that improvement separately as well.

Reply:
Tying to address it in the same PR.